python-frontend/src/main/java/org/sonar/python/frontend/PythonParser.java

```diff
@@ -63,7 +70,28 @@
   private static final PsiFileFactory psiFileFactory = psiFileFactory();
 
   public PyFile parse(String content) {
-    return (PyFile) psiFileFactory.createFileFromText("test.py", PythonFileType.INSTANCE, normalizeEol(content), System.currentTimeMillis(), false, false);
+    PyFile file = parseAs(content, LanguageLevel.PYTHON38);
+    if (errorElements(file).isEmpty()) {
+      return file;
+    }
+    file = parseAs(content, LanguageLevel.PYTHON27);
+    PsiErrorElement errorElement = errorElements(file).get(0);
+    if (errorElement == null) {
+      return file;
+    }
+    int lineNumber = new PythonTokenLocation(errorElement).startLine();
+    throw new RecognitionException(lineNumber, errorElement.getErrorDescription());
+  }
+
+  private JBIterable<PsiErrorElement> errorElements(PsiElement root) {
```

Comment:
There are two code smells regarding `errorElements` and `parseAs` not being static

Reply:
Fixed.